test/decl/protocol/special/coding/enum_codable_no_associated_values.swift

```diff
@@ -0,0 +1,14 @@
+// RUN: %target-typecheck-verify-swift -verify-ignore-unknown
+
+// cases without associated values should not generate separate CodingKeys
+// implementations, but use a shared one instead.
+enum SimpleEnum : Codable {
+  case x
+  case y
+  case z
+}
+
+let _ = SimpleEnum.XCodingKeys.self // expected-error {{type 'SimpleEnum' has no member 'XCodingKeys'}}
+let _ = SimpleEnum.YCodingKeys.self // expected-error {{type 'SimpleEnum' has no member 'YCodingKeys'}}
+let _ = SimpleEnum.ZCodingKeys.self // expected-error {{type 'SimpleEnum' has no member 'ZCodingKeys'}}
```

Comment:
I think we need to make these into `typealiase`s as it is otherwise source breaking.

Reply:
Good point!

Reply:
Updated